File: src/main/java/ru/ezhov/rocket/action/types/gist/GistActionService.java
package ru.ezhov.rocket.action.types.gist;

import org.eclipse.egit.github.core.Gist;
import org.eclipse.egit.github.core.client.GitHubClient;
import org.eclipse.egit.github.core.service.GistService;
import ru.ezhov.rocket.action.api.RocketActionSettings;

import java.io.IOException;
import java.util.List;

public class GistActionService {
    public static final String TOKEN_PROPERTY = "rocket.action.gist.token";

    public List<Gist> gists(RocketActionSettings settings) throws GistActionServiceException {
        GitHubClient gitHubClient = new GitHubClient();
        String token = settings.settings().get(GistRocketAction.TOKEN);
        if (token == null) {
            token = System.getProperty(TOKEN_PROPERTY);
        }
        gitHubClient.setOAuth2Token(token);
        GistService gistService = new GistService(gitHubClient);
        String username = settings.settings().get("username");
        try {
            return gistService.getGists(username);
        } catch (IOException e) {
            throw new GistActionServiceException("Error catch list Gist for user '" + username + "'", e);
        }
    }
}

File: src/main/kotlin/ru/ezhov/rocket/action/types/gist/GistActionService.kt
package ru.ezhov.rocket.action.types.gist

import org.eclipse.egit.github.core.Gist
import org.eclipse.egit.github.core.client.GitHubClient
import org.eclipse.egit.github.core.service.GistService
import java.io.IOException
import java.util.logging.Logger

class GistActionService {
    @Throws(GistActionServiceException::class)
    fun gists(token: String, username: String): List<Gist> {
        val gitHubClient = GitHubClient()
        gitHubClient.setOAuth2Token(token)
        val gistService = GistService(gitHubClient)

        return try {
            gistService.getGists(username)
        } catch (e: IOException) {
            throw GistActionServiceException("Error catch list Gist for user '$username'", e)
        }
    }

    companion object {
        private val LOGGER = Logger.getLogger(GistActionService::class.java.name)
    }
}